lib/SyTest/Homeserver/Synapse.pm

```diff
@@ -896,6 +872,27 @@ backend client_reader
 EOCONFIG
 }
 
+sub generate_haproxy_map
+{
+    return <<"EOCONFIG";
```

Comment:
Since there's no perl-level variable interpolation in here, you can use `'EOCONFIG'` to be an q()'ed string so $ is not special there.